homework/index.js

```diff
@@ -1,46 +1,99 @@
-      } else {
-        cb(new Error(`Network error: ${xhr.status} - ${xhr.statusText}`));
-      }
-    };
-    xhr.onerror = () => cb(new Error('Network request failed'));
-    xhr.send();
-  }
+function fetchJSON(url, cb) {
+  const xhr = new XMLHttpRequest();
+  xhr.open('GET', url);
+  xhr.responseType = 'json';
+  xhr.onload = () => {
+    if (xhr.status >= 200 && xhr.status <= 299) {
+      cb(null, xhr.response);
+    } else {
+      cb(new Error(`Network error:${xhr.status} -${xhr.statusText}`));
+    }
+  };
+  xhr.onerror = () => cb(new Error('Network request failed'));
+  xhr.send();
+}
 
-  function createAndAppend(name, parent, options = {}) {
-    const elem = document.createElement(name);
-    parent.appendChild(elem);
-    Object.keys(options).forEach(key => {
-      const value = options[key];
-      if (key === 'text') {
-        elem.textContent = value;
-      } else {
-        elem.setAttribute(key, value);
-      }
-    });
-    return elem;
-  }
+function createAndAppend(name, parent, options = {}) {
+  const elem = document.createElement(name);
+  parent.appendChild(elem);
+  Object.keys(options).forEach(key => {
+    const value = options[key];
+    if (key === 'text') {
+      elem.textContent = value;
+    } else {
+      elem.setAttribute(key, value);
+    }
+  });
+  return elem;
+}
+
+function main(url) {
+  fetchJSON(url, (err, repositories) => {
+    const root = document.getElementById('root');
+    if (err) {
+      createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+      return;
+    }
+    const repNames = [];
+    createAndAppend('select', root, { id: 'main-select' });
+    createAndAppend('div', root, { class: 'portion-parent', id: 'parent' });
+    const parent = document.getElementById('parent');
+    createAndAppend('div', parent, { class: 'left-portion', id: 'lft' });
+    createAndAppend('div', parent, { class: 'right-portion', id: 'rt' });
 
-  function main(url) {
-    fetchJSON(url, (err, repositories) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-        return;
+    const leftPortion = document.getElementById('lft');
+    const rightPortion = document.getElementById('rt');
+    for (let i = 0; i < repositories.length; i++) {
+      const rep = repositories[i];
+      repNames.push(rep.name);
+    }
+    const selectItem = document.getElementById('main-select');
+
+    repNames.sort();
+    for (let i = 0; i < repNames.length; i++) {
+      createAndAppend('option', selectItem, {
+        value: i,
+        text: repNames[i],
+      });
+    }
+    selectItem.selectedIndex = 0;
+    selectItem.addEventListener('change', () => {
+      const reponame = selectItem.options[selectItem.selectedIndex].text;
+      for (let i = 0; i < repositories.length; i++) {
+        if (repositories[i].name === reponame) {
+          const repo = repositories[i];
```

Comment:
Because you can access the selected repository object directly by using the index there is no need to loop over all array elements to find the relevant element. You can remove line 64 and 65 and the corresponding closing brace and change line 66 into:

```js
const repo = repositories[select.value];
```